tests/runners/bulk_update_runner.py

```diff
@@ -0,0 +1,138 @@
+        update_spreadsheet, updated_spreadsheet_path = self.__download_modified_spreadsheet()
+        updated_project_title, updated_biomaterial_name, modified_biomaterial_id = \
+            self.__modify_metadata_in_sheet(update_spreadsheet)
+        self.bulk_update_manager.save_modified_spreadsheet(update_spreadsheet, updated_spreadsheet_path)
+        self.__upload_modified_spreadsheet(updated_spreadsheet_path)
+        self.__validate_modifications(updated_project_description, updated_contributor_name, updated_bsd_accession,
+                                      updated_project_title, updated_biomaterial_name)
+
+    def __import_spreadsheet_with_files(self, dataset_fixture, project_uuid=None):
+        self.dataset = dataset_fixture
+        self.__upload_spreadsheet_and_create_submission(dataset_fixture, project_uuid)
+        self.submission_manager = SubmissionManager(self.submission_envelope)
+        self.submission_manager.get_upload_area_credentials()
+        self.submission_manager.stage_data_files(self.dataset.config['data_files_upload_area_uuid'])
+        self.submission_manager.wait_for_envelope_to_be_validated()
+
+    def __upload_spreadsheet_and_create_submission(self, dataset_fixture, project_uuid=None):
+        spreadsheet_filename = os.path.basename(dataset_fixture.metadata_spreadsheet_path)
+        Progress.report(f"CREATING SUBMISSION with {spreadsheet_filename}...")
+        self.submission_id = self.ingest_broker.upload(dataset_fixture.metadata_spreadsheet_path,
+                                                       project_uuid=project_uuid)
+        Progress.report(f"submission is in {self.ingest_api.ingest_api_url}/submissionEnvelopes/{self.submission_id}\n")
+        self.submission_envelope = self.ingest_api.envelope(self.submission_id)
+
+    def __upload_modified_spreadsheet(self, path_to_spreadsheet):
+        self.ingest_broker.upload(path_to_spreadsheet, is_update=True)
+        self.submission_manager.wait_for_envelope_to_be_validated()
+
+    def __get_original_content(self):
+        project_payload = self.__get_project_content()
+        self.project_id = self.bulk_update_manager.get_id_from_entity(project_payload)
+
+        self.__get_biomaterial_content_by_id()
+        self.biomaterial_ids = [*self.biomaterial_content_by_id]
+
+    def __get_project_content(self):
+        project_payload = \
+            self.bulk_update_manager.get_entities_by_submission_id_and_type(self.submission_id, 'projects')[0]
+        self.project_content = project_payload.get('content')
+        return project_payload
+
+    def __get_biomaterial_content_by_id(self):
+        biomaterials_payload = \
+            self.bulk_update_manager.get_entities_by_submission_id_and_type(self.submission_id, 'biomaterials')
+        self.biomaterial_content_by_id = {
+            self.bulk_update_manager.get_id_from_entity(payload): payload.get('content')
+            for payload in biomaterials_payload
+        }
```

Comment:
interesting syntax, i didn't know you could do this

Reply:
Well, sometimes you can even learn from me, too 😜